### PR TITLE
Add less support

### PR DESCRIPTION
**Pull request by onedesert**
This PR closes https://github.com/AtomLinter/linter-stylelint/issues/135

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Can you add a spec to check that it properly enables the syntax on those files? See https://github.com/AtomLinter/linter-stylelint/pull/149#issuecomment-210891344 for an outline on how to do this.

**Comment by onedesert** [comment]
@Arcanemagus can you review it please? and let me know if it's ok? if it's so I could do the same for scss files and postcss

**Comment by Arcanemagus** [comment]
Looking

File: spec/fixtures/less/good.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.div {
```
Comment (Arcanemagus) [inline spec/fixtures/less/good.css]: Shouldn't these files be `.less`?

Comment (onedesert) [inline spec/fixtures/less/good.css]: Yes, thanks! sorry

File: spec/linter-stylelint-spec.js [hunk 2]
```diff
@@ -171,4 +196,5 @@ describe('The stylelint provider for Linter', () => {
       })
     );
   });
+
```
Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: No reason for this blank line... there isn't another block after this.

Comment (onedesert) [inline spec/linter-stylelint-spec.js]: ab6624d

**Comment by Arcanemagus** [comment]
The test files should probably be explicitly LESS files, meaning they are actually utilizing whatever it is that LESS adds on top of CSS.

So `good.less` should be invalid CSS (if that's the way it works), this way we would know for sure that the files are properly being processed as LESS and not just CSS.

File: spec/fixtures/less/good.css (hunk 1 above)
Comment (onedesert) [inline spec/fixtures/less/good.css]: 423d3c5, e7abf13

File: spec/linter-stylelint-spec.js (hunk 2 above)
Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: Heh, this was actually the cause of the build failures as this wasn't passing the lint checks :stuck_out_tongue:.

File: spec/linter-stylelint-spec.js [hunk 3]
```diff
@@ -4,8 +4,11 @@ import * as path from 'path';
 
 const badDir = path.join(__dirname, 'fixtures', 'bad');
 const configStandardPath = path.join(badDir, 'stylelint-config-standard.css');
+const badLessDir = path.join(__dirname, 'fixtures', 'less');
```
Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: Rename this to `lessDir`.

**Comment by onedesert** [comment]
good.less now invalid css in db9592a. @Arcanemagus can I ask you how to test locally? it's the first time I test an atom package and `apm test` is always passing.

File: spec/linter-stylelint-spec.js [hunk 4]
```diff
@@ -4,8 +4,11 @@ import * as path from 'path';
...
+const configStandardLessPath = path.join(badLessDir, 'stylelint-config-standard.less');
 const warn = path.join(__dirname, 'fixtures', 'warn', 'warn.css');
 const good = path.join(__dirname, 'fixtures', 'good', 'good.css');
+const goodLess = path.join(__dirname, 'fixtures', 'less', 'good.less');
```
Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: Use the directory variable you specified above, no reason to have two different specifications for the same folder.

File: spec/linter-stylelint-spec.js (hunk 2 above)
Comment (onedesert) [inline spec/linter-stylelint-spec.js]: My bad focused on test I did'n check for code linting

File: spec/linter-stylelint-spec.js (hunk 3 above)
Comment (onedesert) [inline spec/linter-stylelint-spec.js]: 5d9ae36

**Comment by Arcanemagus** [comment]
I just run `linter-eslint` and don't commit files that have messages showing from it, the CI builds run `npm test` which is an alias of `./node_modules/.bin/eslint .` which runs `eslint` over the files and checks for style/code issues.

File: spec/linter-stylelint-spec.js (hunk 4 above)
Comment (onedesert) [inline spec/linter-stylelint-spec.js]: 591cbe4

File: spec/linter-stylelint-spec.js [hunk 5]
```diff
@@ -4,8 +4,11 @@ import * as path from 'path';
 
 const badDir = path.join(__dirname, 'fixtures', 'bad');
 const configStandardPath = path.join(badDir, 'stylelint-config-standard.css');
+const lessDir = path.join(__dirname, 'fixtures', 'less');
+const configStandardLessPath = path.join(lessDir, 'stylelint-config-standard.less');
 const warn = path.join(__dirname, 'fixtures', 'warn', 'warn.css');
 const good = path.join(__dirname, 'fixtures', 'good', 'good.css');
+const goodLess = path.join(__dirname, 'fixtures', 'less', 'good.less');
```
Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: Looks like the earlier comment saying this got "closed", this should be:

```js
const goodLess = path.join(lessDir, 'good.less');
```

Comment (Arcanemagus) [inline spec/linter-stylelint-spec.js]: Lol, and you fixed it as I commented that :+1:.

**Comment by Arcanemagus** [comment]
LGTM at this point, assuming CI doesn't complain.

**Comment by onedesert** [comment]
Perfect, I hope it has been useful. Sorry for all this issues it's the first time I do something with an atom package. I've done only because we move today to use stylelint as now supports less and it's good to have this linter to fix 500 errors. Thanks for this cool package!

**Comment by Arcanemagus** [comment]
Thanks for putting this together, it's always nice to see people getting involved :grinning:.

**Comment by Arcanemagus** [comment]
Merging, thanks again! (That intermediate failure in AppVeyor looks like a network issue, started that rebuilding just because)

**Comment by Arcanemagus** [comment]
Waiting on #149 to release this as it looks like that will probably finish up today as well.

**Comment by onedesert** [comment]
Yes looks so. If https://github.com/AtomLinter/linter-stylelint/pull/149 need any help with conflicts let me know.

**Comment by Arcanemagus** [comment]
Published in v2.4.0 :tada:!

**Pull request merged**